packages/xstate-react/src/index.ts

```diff
@@ -93,13 +86,22 @@ export function useMachine<TContext, TEvent extends EventObject>(
   }, [actions]);
 
   // Start service immediately (before mount) if specified in options
-  if (immediate) {
```

Comment:
why a simple reordering of this if statements and following `useState` wouldnt work?

Reply:
You're right! That was actually the first thing I tried, and at the time I thought it had introduced an issue where the action associated with the null transition was re-executed, but after trying it again I no longer see that occurring.

I'll update this PR to a simple re-ordering of those statements.